calculator.py

```diff
@@ -23,7 +23,16 @@ def execute(operation,x,y):    # function to execute operation
 
 def get_number(position, repeat=True):     # function to request for user's input
     
+    count = 0
     while repeat:
+
+        # track and return trials exceeded after 5 times
+        count += 1
+        if count == 6:
+            print("\n5 Trials Exceeded!!! Program Ended\n\n.... Try again later and enter a number\n")
+            return "Trials Exceeded"
+        
+        # ask user for number
         number = input("Enter the {}: ".format(position))
 
         try:
```

Comment:
this code block looks redundant.
You can try cast the input in line 36.

here is a cleaner approach that eliminates the while loop and provides better error handling

def get_number(position, max_trials=5):
    for _ in range(max_trials):
        try:
            number = float(input(f"Enter the {position}: "))
            return number
        except ValueError:
            print("Invalid input. Please enter a valid number.")
    print(f"{max_trials} Trials Exceeded! Program Ended.")
    return None

Reply:
Thanks! I will make the corrections right away.